processor/src/main/java/org/mapstruct/ap/model/source/IterableMapping.java

```diff
@@ -37,11 +40,21 @@
     private final AnnotationMirror mirror;
     private final AnnotationValue dateFormatAnnotationValue;
 
-    public static IterableMapping fromPrism(IterableMappingPrism iterableMapping) {
+    public static IterableMapping fromPrism(IterableMappingPrism iterableMapping, ExecutableElement method,
+                                            Messager messager) {
         if ( iterableMapping == null ) {
             return null;
         }
 
+        if ( iterableMapping.dateFormat().isEmpty() && iterableMapping.qualifiedBy().isEmpty() ) {
+            messager.printMessage(
+                Diagnostic.Kind.ERROR,
+                "'dateformat' and 'qualifiedBy' are both undefined in @IterableMapping, either define a "
```

Comment:
Suggestion: let the message end with ", define at least one of them." - similar to your message for MapMapping. That way it's one spot less to modify in case we add new attributes to that annotation, and it's more consistent.